controllers/providers/metrics/prometheus_test.go

```diff
@@ -175,6 +224,68 @@ func TestUpgradeIngressHost(t *testing.T) {
 	assert.Equal(t, 1., cntNotFound2)
 }
 
+func TestUpdateFailover(t *testing.T) {
+	// arrange
+	m := newPrometheusMetrics(defaultConfig)
+
+	// act
+	m.UpdateFailoverStatus(defaultGslb, true, k8gbv1beta1.Healthy, []string{"10.0.0.1", "10.0.0.2"})
+	// assert
+	hp := testutil.ToFloat64(m.Get(K8gbGslbFailoverStatus).AsGaugeVec().
+		With(prometheus.Labels{"namespace": namespace, "name": gslbName, "status": fmt.Sprintf("%s_%s", k8gbv1beta1.Healthy, primary)}))
+	up := testutil.ToFloat64(m.Get(K8gbGslbFailoverStatus).AsGaugeVec().
+		With(prometheus.Labels{"namespace": namespace, "name": gslbName, "status": fmt.Sprintf("%s_%s", k8gbv1beta1.Unhealthy, primary)}))
+	fp := testutil.ToFloat64(m.Get(K8gbGslbFailoverStatus).AsGaugeVec().
+		With(prometheus.Labels{"namespace": namespace, "name": gslbName, "status": fmt.Sprintf("%s_%s", k8gbv1beta1.NotFound, primary)}))
```

Comment:
@kuritka sorry, if I am missing something obvious, but how `targets` are actually used here? I see they come as an argument to `UpdateFailoverStatus` but it's not clear how they actually propagated

Reply:
@ytsarev , good question. 
![Screenshot 2021-08-24 at 16 05 48](https://user-images.githubusercontent.com/7195836/130631231-1259b69f-6899-47d1-b28b-a5358dac9277.png)

failover status is gaugevec, divided by status into healthy, unhealthy, not found. Each status has a value that changes in time depending on the state of the targets. In the case, when the UpdateFailoverStatus function is called, all three statuses are set according to the status of targets. For a while it's healthy 2, for a while it's 0, but unhealthy 2, etc... 


The place where targets are passed to metrics is here: https://github.com/k8gb-io/k8gb/blob/prometheus-metrics-add/controllers/dnsupdate.go#L106